src/wallet/wallet.h

```diff
@@ -798,10 +800,10 @@ class CWallet final : public FillableSigningProvider, private interfaces::Chain:
      * Abandoned state should probably be more carefully tracked via different
      * posInBlock signals or by checking mempool presence when necessary.
      */
-    bool AddToWalletIfInvolvingMe(const CTransactionRef& tx, CWalletTx::Status status, const uint256& block_hash, int posInBlock, bool fUpdate) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
+    bool AddToWalletIfInvolvingMe(const CTransactionRef& tx, CWalletTx::Status status, const uint256& block_hash, int posInBlock, bool fUpdate, int block_height) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
```

Comment:
In commit "Add block_height field in struct Confirmation" (de8fe7d717042823f2aa429be2a260904a10b42e)

Could you move the block_height argument after block_hash to be consistent with `SyncTransaction` argument order and `Confirmation` struct order?

Alternately, it possible, it would be really nice to combine the 4 status/hash/height/index arguments into a single CWalletTx::Confirmation struct argument, because having 4 int arguments in a row makes bugs more likely.

Reply:
See d0c4ce6